Nodejs/Product/TestAdapter/ProjectTestDiscoverer.cs

```diff
@@ -12,11 +12,12 @@
 
 namespace Microsoft.NodejsTools.TestAdapter
 {
-    // Keep in sync the method TypeScriptHelpers.IsSupportedTestProjectFile if there's a change on the supported projects.
-    [FileExtension(NodejsConstants.NodejsProjectExtension), FileExtension(NodejsConstants.CSharpProjectExtension), FileExtension(NodejsConstants.VisualBasicProjectExtension), FileExtension(NodejsConstants.JavaScriptProjectExtension)]
-    [DefaultExecutorUri(NodejsConstants.ExecutorUriString)]
-    public partial class ProjectTestDiscoverer : ITestDiscoverer
+    // We require to put a non-existent file extension to avoid duplicate discovery executions.
+    [FileExtension("NTVS_NonExistentFileExtension")]
+    public class ProjectTestDiscoverer : ITestDiscoverer
```

Comment:
`abstract` ?

Reply:
This class is instantiated in: https://github.com/microsoft/nodejstools/blob/587d04559aaef7288c51ef8420fff67c38257006/Nodejs/Product/TestAdapter/ProjectTestExecutor.cs#L30

This happens when the users launched execution from the vstest cli.